R/test-files.r

```diff
@@ -23,34 +23,78 @@ test_env <- function() {
 #'   regular expression will be executed.  Matching will take on the file
 #'   name after it has been stripped of \code{"test-"} and \code{".r"}.
 #' @param env environment in which to execute test suite.
-#' @return a data frame of the summary of test results
+#' @return the results as a "testthat_results" (list)
 #' @export
 test_dir <- function(path, filter = NULL, reporter = "summary",
-                     env = test_env()) {
+                                          env = test_env()) {
+
   current_reporter <- find_reporter(reporter)
-  lister <- ListReporter$new()
-  reporter <- MultiReporter$new(reporters = list(current_reporter, lister))
+  source_test_helpers(path, env)
+  paths <- find_test_scripts(path, filter)
+  if (length(paths) == 0) stop('No matching test file in dir')
+
+  current_reporter$start_reporter()
+  results <- lapply(paths, .test_file, parent_env = env,
+    reporter = current_reporter, start_end_reporter = FALSE)
+  current_reporter$end_reporter()
 
+  results <- unlist(results, recursive = FALSE)
+
+  invisible(structure(results, class = 'testthat_results'))
+}
+
+
+#' source the helper scripts if any
```

Comment:
Ok, maybe flag as `@keywords internal` so the don't show up in the main doc index?